spine-c/src/spine/Skeleton.c

```diff
@@ -234,6 +234,19 @@ void spSkeleton_setSlotsToSetupPose (const spSkeleton* self) {
 		spSlot_setToSetupPose(self->slots[i]);
 }
 
+void spSkeleton_addBone (spSkeleton* self, spBone *bone) {
+	self->bones = (spBone **) realloc(self->bones, sizeof(spBone *) * (self->bonesCount + 1));
+	self->bones[self->bonesCount] = bone;
+
+	self->data->bones = (spBoneData **) realloc(self->data->bones, sizeof(spBoneData *) * (self->bonesCount + 1));
+	self->data->bones[self->bonesCount] = bone->data;
+
+	++self->bonesCount;
+	++self->data->bonesCount;
+
+	spSkeleton_updateCache(self);
```

Comment:
Expensive, should not be called every time a bone is added. Require user to call it.